src/common/AndroidUtils.ts

```diff
@@ -558,13 +558,20 @@ export class AndroidUtils {
      * @param waitForBoot Optional boolean indicating whether it should wait for the device to boot up. Defaults to true.
      */
     public static async rebootEmulator(portNumber: number, waitForBoot: boolean, logger?: Logger): Promise<void> {
-        return AndroidUtils.executeAdbCommand('shell reboot', portNumber, logger).then(() => {
-            if (waitForBoot) {
-                return AndroidUtils.waitUntilDeviceIsReady(portNumber, logger);
-            } else {
-                return Promise.resolve();
-            }
-        });
+        try {
```

Comment:
Our `reboot` command for Android has been broken for a while now (not sure exactly when it broke). `adb` sometimes returns an error even though it does reboot the device.... I changed the logic here to be more robust in rebooting.